File: src/main/java/com/miti99/storescraperbot/api/apple/request/AppleAppRequest.java
package com.miti99.storescraperbot.api.apple.request;

public record AppleAppRequest(Long id, String appId, String country, Boolean ratings) {

  public AppleAppRequest(Long id, String country) {
    this(id, null, country, true);
  }

  public AppleAppRequest(String appId, String country) {
    this(null, appId, country, true);
  }
}

File: internal/api/apple/request/apple_app_request.go
package request

// AppleAppRequest mirrors Java AppleAppRequest record. Either ID (iTunes
// trackId) or AppID (bundleId) is set; the other is omitted from JSON.
type AppleAppRequest struct {
	ID      *int64  `json:"id,omitempty"`
	AppID   *string `json:"appId,omitempty"`
	Country string  `json:"country"`
	Ratings bool    `json:"ratings"`
}

func ByTrackID(id int64, country string) AppleAppRequest {
	return AppleAppRequest{ID: &id, Country: country, Ratings: true}
}

func ByBundleID(appID, country string) AppleAppRequest {
	return AppleAppRequest{AppID: &appID, Country: country, Ratings: true}
}
